### PR TITLE
✨ TokenEnitity on RMRK

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot Indexer](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] Related With https://github.com/kodadot/nft-gallery/issues/5385

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

### Optional

- [x] I've run the indexer and it hasn't failed
- [x] I've changed schema and performed a migration
- [ ] I found edge cases

### Screenshot

- [ ] My contribution is a **resolver** so I have pic from playground

@vikiival

### CODE REVIEW

File: src/mappings/v2/mint.ts [hunk 1]
```diff
@@ -9,9 +9,10 @@ import { CollectionEntity, NFTEntity, Property } from '../../model/generated'
 import { createEvent } from '../shared/event'
 import { handleMetadata, isLewd } from '../shared/metadata'
 import { findRootItemById } from '../utils/entity'
-import { calculateCollectionOwnerCountAndDistribution, isDummyAddress } from '../utils/helper'
+import {  isDummyAddress } from '../utils/helper'
```
Comment (roiLeo) [inline src/mappings/v2/mint.ts:12]:
```suggestion
import { isDummyAddress } from '../utils/helper'
```

**Review by roiLeo — COMMENTED** [review]
does it run?

**Comment by daiagi** [comment]
> does it run?

wdym?

 I tested it on local node and it works

**Comment by roiLeo** [comment]
> I tested it on local node and it works

could you share (if possible) a screenshot of a query result on tokenEntity?

**Comment by daiagi** [comment]
> > I tested it on local node and it works
> 
> could you share (if possible) a screenshot of a query result on tokenEntity?

Sure


image

**Review by roiLeo — APPROVED** [review]

File: src/mappings/shared/handleTokenEntity.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  const nftMedia =
+    nft.image ??
+    nft.media ??
+    nft.meta?.image ??
+    nft.meta?.animationUrl ??
+    nft.resources[0].src ??
+    nft.resources[0].thumb
+  if (!nftMedia || nftMedia === '') {
+    warn(OPERATION, `MISSING NFT MEDIA ${nft.id}`)
+    await new Promise((resolve) => setTimeout(resolve, 5000))
```
Comment (vikiival) [inline src/mappings/shared/handleTokenEntity.ts:19]: Why are you waiting 5 sec here?

File: src/mappings/v1/mint.ts [hunk 3]
```diff
@@ -69,6 +70,11 @@ export async function mintItem(context: Context): Promise<void> {
       }
     }
 
+    const token = await handleTokenEntity(context, collection, final)
+    if (token) {
+      final.token = token
+    }
```
Comment (vikiival) [inline src/mappings/v1/mint.ts:76]: handleTokenEntity can return `Promise<null | TokenEntity>` and you do not need if statement

**Review by vikiival — CHANGES_REQUESTED** [review]
the five sec delay makes it slow, please clarify ^-^

File: src/mappings/shared/handleTokenEntity.ts (hunk 2 above)
Comment (daiagi) [reply src/mappings/shared/handleTokenEntity.ts:19]: oops sorry. it was for debugging, i wanted to make the log slow down.
of course it doesn't belong in production

Comment (daiagi) [reply src/mappings/shared/handleTokenEntity.ts]: removed

File: src/mappings/v1/mint.ts (hunk 3 above)
Comment (daiagi) [reply src/mappings/v1/mint.ts]: done

**Review by vikiival — APPROVED** [review]

**Pull request merged**